flash/vision/classification/data.py

```diff
@@ -168,56 +150,40 @@ def collate(self, samples: Sequence) -> Any:
-        return self.common_pre_tensor_transform(source, self.test_transform), target
-
-    def predict_pre_tensor_transform(self, sample: Any) -> Any:
+    def common_step(self, sample: Any) -> Any:
+        if isinstance(sample, (list, tuple)):
+            source, target = sample
+            return self.current_transform(source), target
+        return self.current_transform(sample)
+
+    def pre_tensor_transform(self, sample: Any) -> Any:
+        return self.common_step(sample)
+
+    def to_tensor_transform(self, sample: Any) -> Any:
+        if self.current_transform == self._identify:
+            if isinstance(sample, (list, tuple)):
+                source, target = sample
+                if isinstance(source, torch.Tensor):
+                    return source, target
+                return self.to_tensor(source), target
+            elif isinstance(sample, torch.Tensor):
+                return sample
+            return self.to_tensor(sample)
         if isinstance(sample, torch.Tensor):
             return sample
-        return self.common_pre_tensor_transform(sample, self.predict_transform)
-
-    def to_tensor_transform(self, sample) -> Any:
-        source, target = sample
-        return source if isinstance(source, torch.Tensor) else self.to_tensor(source), target
+        return self.common_step(sample)
 
-    def predict_to_tensor_transform(self, sample) -> Any:
-        if isinstance(sample, torch.Tensor):
-            return sample
-        return self.to_tensor(sample)
+    def post_tensor_transform(self, sample: Any) -> Any:
+        return self.common_step(sample)
 
-    def common_post_tensor_transform(self, sample: Any, transform) -> Any:
-        return self._apply_transform(sample, transform, "post_tensor_transform")
+    def per_batch_transform(self, sample: Any) -> Any:
+        return self.common_step(sample)
 
-    def train_post_tensor_transform(self, sample: Any) -> Any:
-        source, target = sample
-        return self.common_post_tensor_transform(source, self.train_transform), target
+    def per_sample_transform_on_device(self, sample: Any) -> Any:
+        return self.common_step(sample)
 
-    def val_post_tensor_transform(self, sample: Any) -> Any:
-        source, target = sample
-        return self.common_post_tensor_transform(source, self.val_transform), target
-
-    def test_post_tensor_transform(self, sample: Any) -> Any:
-        source, target = sample
-        return self.common_post_tensor_transform(source, self.test_transform), target
-
-    def predict_post_tensor_transform(self, sample: Any) -> Any:
-        return self.common_post_tensor_transform(sample, self.predict_transform)
-
-    def train_per_batch_transform_on_device(self, batch: Tuple) -> Tuple:
-        batch, target = batch
-        return self._apply_transform(batch, self.train_transform, "per_batch_transform_on_device"), target
+    def per_batch_transform_on_device(self, sample: Any) -> Any:
```

Comment:
same question as before, IMO this should return tensor type

Reply:
It can be any DataStructure defined by the users.

Reply:
can be container of tensors